test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/group/OrganisationListControllerSpec.scala

```diff
@@ -0,0 +1,152 @@
+
+package uk.gov.hmrc.plasticpackagingtax.registration.controllers.group
+
+import base.unit.ControllerSpec
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, when}
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.test.Helpers.{redirectLocation, status}
+import play.twirl.api.HtmlFormat
+import uk.gov.hmrc.plasticpackagingtax.registration.models.registration.GroupDetail
+import uk.gov.hmrc.plasticpackagingtax.registration.views.html.group.organisation_list
+import uk.gov.hmrc.plasticpackagingtax.registration.{controllers => pptControllers}
+import uk.gov.hmrc.play.bootstrap.tools.Stubs.stubMessagesControllerComponents
+
+class OrganisationListControllerSpec extends ControllerSpec {
+  private val page = mock[organisation_list]
+  private val mcc  = stubMessagesControllerComponents()
+
+  private val controller =
+    new OrganisationListController(authenticate = mockAuthAction,
+                                   mockJourneyAction,
+                                   mockRegistrationConnector,
+                                   mcc = mcc,
+                                   page = page
+    )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    when(page.apply(any(), any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+
+    val registration = aRegistration(
+      withGroupDetail(
+        Some(GroupDetail(membersUnderGroupControl = Some(true), members = Seq(groupMember)))
+      )
+    )
+    mockRegistrationFind(registration)
+  }
+
+  override protected def afterEach(): Unit = {
+    reset(page)
+    super.afterEach()
+  }
+
+  "Organisation list controller" should {
+
+    "return 200" when {
+
+      "user is authorised and display page method is invoked" in {
+        authorizedUser()
+
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe OK
+      }
+    }
+
+    "return 303" when {
+
+      "no group members exist" in {
+        authorizedUser()
+        mockRegistrationFind(aRegistration())
+        val result = controller.displayPage()(getRequest())
+
+        status(result) mustBe SEE_OTHER
+        redirectLocation(result) mustBe Some(
+          pptControllers.routes.RegistrationController.displayPage().url
+        )
+      }
+
+    }
+
+    "return 400" when {
+
+      "user does not make a selection" in {
+        authorizedUser()
+
+        val correctForm = Seq("addOrganisation" -> "", saveAndContinueFormAction)
+        val result      = controller.submit()(postJsonRequestEncoded(correctForm: _*))
+
+        status(result) mustBe BAD_REQUEST
+
+      }
+
+    }
+
+    "return an error" when {
+
+      "user is not authorised" in {
+        unAuthorizedUser()
+        val result = controller.displayPage()(getRequest())
+
+        intercept[RuntimeException](status(result))
+      }
+    }
+
+    "redirects to registration page" when {
```

Comment:
task list?

Reply:
I'll re-organise these tests into "display" and "submit" tests.  There should be one display direct when the GroupDetails aren't present and a different one when the GroupMembers list is empty.